src/parser.ts

```diff
@@ -306,7 +310,8 @@ export class Parser {
       this._t.next();
     }
 
-    return this.finish({ name: 'text', contents });
+    let endLoc = !this._posStack || lastRealTok === null ? undefined : lastRealTok.location!.end;
```

Comment:
```suggestion
    let endLoc = this._posStack && lastRealTok?.location?.end;
```

Reply:
> `lastRealTok?.location?.end`

The `!` is fine after `location` because it really is always defined if it's reached - `this._posStack` being truthy implies every token has a `location` property.

... Well, that's what I thought, but using `lastRealTok?.location!.end` breaks tests. Which is surprising to me; I had thought the `!` was just a type assertion, but it appears to be changing the semantics of the generated code. I guess TS is parsing it as the end of the "chain"? That's _really_ surprising to me.

... Oh, I see this has already been [discovered](https://github.com/microsoft/TypeScript/issues/36031) and [fixed](https://github.com/microsoft/TypeScript/pull/36539) upstream. I guess I'll just `// @ts-ignore` this line until that's released.